src/p2p/net_node.inl

```diff
@@ -1555,14 +1555,14 @@ namespace nodetool
           if (!m_fallback_seed_nodes_added)
           {
             MWARNING("Failed to connect to any of seed peers, trying fallback seeds");
-            current_index = m_seed_nodes.size();
+            current_index = m_seed_nodes.size() - 1;
```

Comment:
I don't see why this matters ... ? Is this purely for aesthetics? The `++current_index >= m_seed_nodes.size()` resets to `0` either way and the conditional check doesn't change.

Reply:
The ++current_index >= m_seed_nodes.size() condition would not fire as the call to append_net_address may add to m_seed_nodes. So the current_index does not get set to 0. Since it was walking through the seeds in random order, it needs to get set to the end of the array to walk through the newly added seed nodes. Due to the ++ to check for loop end, setting it to m_seed_nodes.size() would cause the first newly added seed node to be skipped.